cpp/src/barretenberg/plonk/composer/composer_base.hpp

```
@@ -107,6 +107,7 @@ class ComposerBase {
     virtual ~ComposerBase(){};
 
     virtual size_t get_num_gates() const { return num_gates; }
+    virtual size_t get_total_circuit_size() const { return num_gates; };
```

This default behavior here is misleading. I will remove that and implement as appropriate in the different composers.